src/core/SampleDecoder.cpp

```diff
@@ -116,15 +116,46 @@ auto decodeSampleDS(const QString& audioFile) -> std::optional<SampleDecoder::Re
 #ifdef LMMS_HAVE_OGGVORBIS
 auto decodeSampleOggVorbis(const QString& audioFile) -> std::optional<SampleDecoder::Result>
 {
-	auto vorbisFile = OggVorbis_File{};
-	const auto openError = ov_fopen(audioFile.toLocal8Bit(), &vorbisFile);
+	static auto s_read = [](void* buffer, size_t size, size_t count, void* stream) -> size_t {
+		auto file = static_cast<QFile*>(stream);
```

Comment:
Maybe out of scope, but since this is core should we try to not use `QFile` here?

Reply:
`QFile` is used to seamlessly handle incompatible file paths between Windows and other POSIX systems. Windows uses UTF-16 for encoding their file paths, while most APIs and POSIX use UTF-8, so if we hadn't used something like `QFile`, we would need to manually convert the file path to a valid UTF-16 encoding for Windows then call some wide string variant of `fopen`, which I don't even think Vorbisfile has.

Instead, it is much easier if we abstract away from the handling of file paths on different systems and just work through `QFile` instead by opening the file there and then using that.

Reply:
So pain later when removing it from core. :)

Reply:
> So pain later when removing it from core. :)

Do you know of any cleaner alternative?

Reply:
Clean isn't the issue here, it's dependency management. We can't leave this in core in the long term.

Reply:
> Clean isn't the issue here, it's dependency management. We can't leave this in core in the long term.

Yes, but `QFile` is the most straightforward solution as of right now. The plan is to move to `std::filesystem` once our CI fully supports it, so we can use that then, and is what I would suggest. At that point, we wouldn't be so dependent on `QFile` and can provide our own compatability layer for file paths on Windows with other systems.

But we don't have `std::fillesystem` in its entirety yet, only in its experimental form, and I don't want to raise complexity by adding experimental and incomplete support for something we will remove anways in place of the full support, which we will do very soon anyways.

Reply:
That's why I asked with `try` specifically. As long as we have a path forward in the future to replace it that's okay.

Reply:
I will add a TODO comment though so that we can remember to come back and change it.